File: server/src/main/java/com/ecommerce/ecommercerestapi/Entity/UserAddress.java
package com.ecommerce.ecommercerestapi.entity;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "user_address")
public class UserAddress {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

    @ManyToOne(cascade = { CascadeType.PERSIST, CascadeType.MERGE })
    private User user;

    private String address;

    private String city;

    @Column(name = "post_code")
    private String postCode;

    private String country;

    private String phone;
    
    private Boolean active;
}

File: client/src/interfaces/UserAddress.ts
import { User } from "./User";

export interface UserAddressReq {
  id?: number;
  user: {
    id: number;
  };
  address: string;
  active?: boolean;
  city: string;
  country: string;
  phone: string;
  postCode: string;
}

export interface UserAddressRes {
  id?: number;
  user: User;
  address: string;
  active: boolean;
  city: string;
  country: string;
  phone: string;
  postCode: string;
}

export const INITIAL_USER_ADDRESS_REQ = {
  user: {
    id: 0,
  },
  address: "",
  active: true,
  city: "",
  country: "",
  phone: "",
  postCode: "",
  id: 0,
};
